be/src/vec/data_types/data_type_object.cpp

```diff
@@ -121,6 +126,11 @@ char* DataTypeObject::serialize(const IColumn& column, char* buf, int be_exec_ve
     }
     // serialize num of subcolumns
     *reinterpret_cast<uint32_t*>(size_pos) = num_of_columns;
+    if (num_of_columns == 0) {
+        // Make sure the num_of_rows in ColumnObject not lost
+        *reinterpret_cast<uint32_t*>(buf) = column_object.rows();
```

Comment:
1. Why only serialize num_rows when num_of_columns == 0?
2. It may cause error when rolling upgrade be if serialization is changed.

Reply:
fixed